syntax = "proto3";

service UserService {
  rpc CreateUser (CreateUserRequest) returns (UserResponse);
  rpc GetUserById (GetUserByIdRequest) returns (UserResponse);
  rpc UpdateUser (UpdateUserRequest) returns (UserResponse);
  rpc DeleteUser (DeleteUserRequest) returns (DeleteUserResponse);
  rpc ListUsers (ListUsersRequest) returns (ListUsersResponse);
}

message User {
  string id = 1;
  string name = 2;
  string email = 3;
}

message CreateUserRequest {
  string name = 1;
  string email = 2;
}

message GetUserByIdRequest {
  string id = 1;
}

message UpdateUserRequest {
  string id = 1;
  string name = 2;
  string email = 3;
}

message DeleteUserRequest {
  string id = 1;
}

message DeleteUserResponse {
  string name = 1;
}

message ListUsersRequest {}

message ListUsersResponse {
  repeated User users = 1;
}

message UserResponse {
  User user = 1;
}
